gossip/hub/handlers.go

```diff
@@ -119,6 +120,19 @@ func trillianSLRChecker(data, sig []byte) error {
 	return nil
 }
 
+func trillianSMRChecker(data, sig []byte) error {
```

Comment:
This function looks like [VerifySignedMapRoot](https://github.com/google/trillian/blob/master/crypto/verifier.go#L48). Would that be better?

Reply:
`sig` is unused. Is that intentional?

Reply:
The signature gets checked (in `addSignedBlob`) before any of the `submissionCheck` instances get called, so not checking `sig` is intentional. 

I included `sig` in the prototype for the structure checkers to allow for the possibility of a structure check that involving the signature, but given there are three now and none of them use it, I suspect YAGNI.  I'll drop it from the `submissionCheck` signature.